### ISSUE TITLE
Mode 3 driver pathing issues

### ISSUE THREAD

**Issue by SteelRodent** [Courseplay/courseplay#6894]
Did the overload course same way as usual (7-8 waypoints on the field edge with a wait in the middle where it should expect the truck to park) - same setup that normally works fine with the Lexion 8900 on other maps (using the same and different auger wagons, with different tractors), but when I tried loading my Felsbruun save today the mode 3 driver is having an off-day.

FWIW the auger wagon is the Horsch from the Giants Horsch DLC/mod. Map is vanilla Felsbruun on Field 10 in this case (never touched this field before).

On first try, after the combine did the first lap of the headland the mode 3 driver then tailgated the combine for the full length of the field (literally slamming into the back of it), almost like it expected a forage harvester. Then after the first turn the mode 3 driver raced around the right side of the combine while it was turning, through the crop, and parked in front of it. Moved it out of the way and restarted, and it immediately went in front of the combine again. Then for the sake of troubleshooting I dropped the weight on the front of the tractor and restarted, and now it can't find the combine at all.

Instead it just sits there and spams the log with this:

2021-02-23 14:22 :33 [info lp227900] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :35 [info lp228000] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :36 [info lp228100] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :38 [info lp228200] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :40 [info lp228300] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :41 [info lp228400] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...
2021-02-23 14:22 :43 [info lp228500] 939 Vario (#3): No path found to CR10.90 (#2) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...

Stopped the driver and double checked that it's set to start at "unload". Tried moving him to other places on the field before starting, including right behind the combine. Still won't budge.

![20210223143104_1](https://user-images.githubusercontent.com/20990941/108850744-fc8c2100-75e3-11eb-8890-e7f2fbac301d.jpg)

Switched to the Agco combine (only other I got on this map) and set it off from where the NH got to. Restart the mode 3 driver.  When the combine was 50% full the mode 3 driver set off and parked in front of the combine

![20210223145627_1](https://user-images.githubusercontent.com/20990941/108853742-62c67300-75e7-11eb-8950-4dddc8bb9d3b.jpg)

Left this in the log:

2021-02-23 14:48 :00 [info lp317822] 939 Vario (#3): Start unloading, waiting for a combine to call
2021-02-23 14:55 :30 [info lp344614] 939 Vario (#3): There is fruit at the target, disabling fruit avoidance

So I moved the mode 3 driver out of the way and restarted it. At which point it decided to do donuts while saying "drive to combine". Stopped and restarted him, and this time he did find a path to the combine, but... once again tried to unload from behind.

![20210223150237_1](https://user-images.githubusercontent.com/20990941/108854521-4545d900-75e8-11eb-9a72-feb1391ecda3.jpg)

So I backed him off again and tried, but once again it just goes into spamming the log without doing anything.

2021-02-23 15:06 :08 [info lp382614] 939 Vario (#3): Start unloading, waiting for a combine to call
2021-02-23 15:06 :10 [info lp382700] 939 Vario (#3): No path found to IDEAL 9T (#4) in 0 ms, pathfinder failed once, relaxing pathfinder field constraint (3.8)...
2021-02-23 15:06 :11 [info lp382800] 939 Vario (#3): No path found to IDEAL 9T (#4) in 0 ms, pathfinder failed at least twice, trying a path through crop and relaxing pathfinder field constraint...

and so on.

[log.txt](https://github.com/Courseplay/courseplay/files/6029388/log.txt)

**Comment by SteelRodent** [comment]
Just for the sake of it, I switched to the Lapacho save, set the Lexion 8900 going. The set the overload on mode 3 going (same Horsch auger wagon, but with a JD 7R tractor). Worked first time, no problems.

**Comment by Tensuko** [comment]
Is the IDEAL Vanilla or Mod ?

**Comment by Tensuko** [comment]
Oh and is avoid fruit active ? I wonder why the mode3 driver sits in the fruit

**Comment by SteelRodent** [comment]
The Ideal is the vanilla version. And avoid fruit is on.

![20210223175259_1](https://user-images.githubusercontent.com/20990941/108877959-0885dc00-7600-11eb-9054-7dd3b0621f8a.jpg)

**Comment by AUZ001** [comment]
I am so glad to see someone else is having the same problems I'm having.
At least now I know I am not going crazy .. lol

P.S - Even with "Avoid Fruit" is activated in both the header and overloader, the overloader still parks in the fruit :(

**Comment by Tensuko** [comment]
Why did you set an offset ?
I am sure you don't need an offset and if so not that high...
Would explain why it runs into the Combine.
![fsScreen_2021_02_23_20_53_34](https://user-images.githubusercontent.com/27344744/108899884-3cb9c680-7619-11eb-9a14-b01c741e3cb3.png)
I can get it kinda stop at least near the waitpoint.
![fsScreen_2021_02_23_21_00_19](https://user-images.githubusercontent.com/27344744/108900701-3bd56480-761a-11eb-998f-40d8acfddebb.png)

But we already have the Plan to create a System to make our Drivers Stop at certain points more reliable.
Like the Mode1 Driver in 'manual loading' Mode, where the mid of the trailers get placed at the start points, so we have less guessing where to place waitpoints. But thats a bit more work to do...

How you got them to park inside the fruit I don't know.
A Video might help.

**Comment by pvaiko** [comment]
@SteelRodent if you had debug channels 7 and 14 on we may be able to figure out what's going on. But based on what I see, the problem is the offset, the pathfinder can't find a path to the combine because with that offset you moved to target into the combine, so the target point is invalid (that's why it fails after 0 milliseconds). Also explains the bumping into the combine on the headland, instead of driving beside the combine, thanks to the offset, the unloader drives right into the combine.

**Comment by pvaiko** [comment]
I'm seriously considering remove that whole offset setting, or at least limit it to probably +-0.5 m, this is causing so many issues and really hurts our reputation :)

**Comment by TinyCrocodile** [comment]
I dont like removing it. It helps many times solving problems with trailers. Sometimes the auto unload point need a tiny bit of adjustment. For example unloading potato harvester it can result in traffic problems due to trailer height. I give it a tiny bit more distance and it works perfect.

**Comment by Tensuko** [comment]
> or at least limit it to probably +-0.5 m

This but with +/- 2m, not more else it will result into those kind of wrong use like abouve.

**Comment by SteelRodent** [comment]
@pvaiko @Tensuko Not sure why there is an offset; I've never set it (that I know of anyway). Although it's been a long time since I touched this save, I have no idea why there would be one set on this tractor since I don't use plows and other "crooked" stuff like that. I figured since it said "auto" it was supposed to be like that. Will see if I can reset it, and then resow the field to test again.

On a sidenote I've found that the Holmer TerraVariant does need an offset because of the odd shape of its bins - like a meter or two to the left of the path, which also makes unloading a bit tricky to get right with these.

**Comment by Tensuko** [comment]
I am happy to hear that it works for you now (even if not perfect).
There the is still some room for tweaking we know about this, but we worked for mode2/mode3 for so many month and found things here and there and we had to say, ok like this must be enough for now, becouse other things haven been left behind.

So can we take the base Issue as solved for now ?
